### PR TITLE
Update TopBannerView to be expandable and actionable

### PR DESCRIPTION
**Pull request by Ecarrion**
Refers to https://github.com/woocommerce/woocommerce-ios/issues/2464

# Why

The new deprecated stats banner is supposed to be expandable and actionable which is something our current `TopBannerView` does not support.

This PR aims to add such functionality to be able to dismiss the banner.
The implementation of that feature can be found [here](https://github.com/woocommerce/woocommerce-ios/pull/2503)

# How

1. Add a new enum to `TopBannerViewModel` to indicate which type of button we want with it's respective handler.
```swift
enum TopButtonType {
  case chevron(handler: (() -> Void)?)
  case dismiss(handler: (() -> Void)?)
}
```

This allows us to remove the previous `onDismiss` and `onExpandedStateChange` parameters on `TopBannerViewModel`

2. Reorganize the internal view layout to be able to hide the action button when the user wants to collapse the banner.

*PS:*  I've decided to not go into a deeper refactor on this view as it could go out of scope.

# GIF
correct-banners

# Testing steps
Run the app and see that the current banners(stats and products) behave correctly.
- To test the stats banner you need to manage a store running an older WC version or disable the wc-admin using [a plugin](https://wordpress.org/plugins/disable-dashboard-for-woocommerce/)

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Top Banner/TopBannerView.swift [hunk 1]
```diff
@@ -122,19 +111,63 @@ private extension TopBannerView {
...
+    func createIconInformationStackView(with viewModel: TopBannerViewModel) -> UIStackView {
+        let informationStackView = createInformationStackView(with: viewModel)
+        let iconInformationStackView = UIStackView(arrangedSubviews: [iconImageView, informationStackView])
+
+        iconInformationStackView.translatesAutoresizingMaskIntoConstraints = false
+        iconInformationStackView.axis = .horizontal
+        iconInformationStackView.spacing = 16
+        iconInformationStackView.alignment = .leading
+        iconInformationStackView.layoutMargins = .init(top: 16, left: 16, bottom: 16, right: 16)
+        iconInformationStackView.isLayoutMarginsRelativeArrangement = true
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Top Banner/TopBannerView.swift:156]: nit: just curious, was `isLayoutMarginsRelativeArrangement` set to fix some UI issue?

File: WooCommerce/Classes/ViewRelated/Top Banner/TopBannerViewModel.swift [hunk 2]
```diff
@@ -3,30 +3,41 @@ import UIKit
 /// View model for `TopBannerView`.
 ///
 struct TopBannerViewModel {
+
+    enum TopButtonType {
+        case chevron(handler: (() -> Void)?)
+        case dismiss(handler: (() -> Void)?)
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Top Banner/TopBannerViewModel.swift:9]: nit: it looks like we aren't using `dismiss` style (cross icon) anymore, maybe we can remove it for now. @Garance91540 do you know if we might be using a top banner that has "X" dismiss CTA at the top that isn't expandable anymore? (for example, in the old stats banners)

File: WooCommerce/Classes/ViewRelated/Top Banner/TopBannerView.swift (hunk 1 above)
Comment (Ecarrion) [reply WooCommerce/Classes/ViewRelated/Top Banner/TopBannerView.swift:156]: This is to make the content laid-out inside `iconInformationStackView` to respect the provided layout margins.

File: WooCommerce/Classes/ViewRelated/Top Banner/TopBannerViewModel.swift (hunk 2 above)
Comment (Ecarrion) [reply WooCommerce/Classes/ViewRelated/Top Banner/TopBannerViewModel.swift:9]: I'll merge this, and will remove the `dismiss` type in a next PR if/when design confirms :-)

**Pull request merged**